compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1892,27 +1892,25 @@ object Types {
     /** The signature of the last known denotation, or if there is none, the
      *  signature of the symbol
      */
-    protected def computeSignature(implicit ctx: Context): Signature = {
-      val lastd = lastDenotation
-      if (lastd != null) lastd.signature
-      else symbol.asSeenFrom(prefix).signature
-    }
+    protected def computeSignature(implicit ctx: Context): Signature =
+      lastDenotation match
+        case null => symbol.asSeenFrom(prefix).signature
```

Comment:
This one too I guess:
```suggestion
        case null => symbol.asSeenFrom(prefix).initial.signature
```